crates/matrix-sdk-ui/src/timeline/event_handler.rs

```diff
@@ -1085,71 +1085,88 @@ impl<'a, 'o> TimelineEventHandler<'a, 'o> {
                 self.items.push_back(item);
             }
 
-            Flow::Remote {
-                position: position @ TimelineItemPosition::Start { .. },
-                txn_id,
-                event_id,
-                ..
+            Flow::Remote { position: TimelineItemPosition::Start { .. }, event_id, .. } => {
+                if self
+                    .items
+                    .iter()
+                    .filter_map(|ev| ev.as_event()?.event_id())
+                    .any(|id| id == event_id)
+                {
+                    trace!("Skipping back-paginated event that has already been seen");
+                    return;
+                }
+
+                trace!("Adding new remote timeline item at the start");
+
+                let item = self.meta.new_timeline_item(item);
+                self.items.push_front(item);
             }
-            | Flow::Remote {
-                position: position @ TimelineItemPosition::End { .. },
-                txn_id,
-                event_id,
-                ..
+
+            Flow::Remote {
+                position: TimelineItemPosition::End { .. }, txn_id, event_id, ..
             } => {
-                // This block tries to find duplicated events.
-
-                let removed_event_item_id = {
```

Comment:
Can you run this block only if `position` is `TimelineItemPosition::End` please?

Reply:
What do you mean? Are you trying to find a fix for https://github.com/matrix-org/matrix-rust-sdk/pull/4378/commits/3b28809c354ca27cff95875cb0d420d2914616e7?

Reply:
Status update, yes he was trying to find a fix, but people want to do a release so we'll merge the revert and do the proper fix afterwards with a regression test.